servlet.py

```diff
@@ -121,6 +124,49 @@ class BaseHandler(tornado.web.RequestHandler):
     def initialize(self):
         self.callback = self.get_argument('callback', default=None)
 
+    @classmethod
+    def initPairsGraph(cls):
```

Comment:
Aha, now I see that `setupHandler` also actually works on the class, not instance. That makes things clearer, thanks (was wondering earlier how we could have `pairs` shared across requests, while anything changed in `BaseHandler.initialize` was renewed for each request). 

… I suppose setupHandler then could also be a class method?